obspy/io/sac/arrayio.py

```diff
@@ -0,0 +1,637 @@
+# -*- coding: utf-8 -*-
+"""
+Low-level array interface to the SAC file format.
+
+Functions in this module work directly with numpy arrays that mirror the SAC
+format.  The 'primitives' in this module are the float, int, and string header
+arrays, the float data array, and a header dictionary. Convenience functions
+are provided to convert between header arrays and more user-friendly
+dictionaries.
+
+These read/write routines are very literal; there is almost no value or type
+checking, except for byteorder and header/data array length.  File- and array-
+based checking routines are provided for additional checks where desired.
+
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.utils import native_str
+from future.builtins import *
+
+import sys
+import warnings
+
+import numpy as np
+
+from obspy.core.compatibility import from_buffer
+
+from ..sac import header as HD
+from .util import SacIOError, SacInvalidContentError
+from .util import is_valid_enum_int, is_same_byteorder, byteswap
+
+
+def init_header_arrays(*args):
+    """
+    Initialize arbitrary header arrays.
+
+    Parameters
+    ----------
+    which : str, {'float', 'int', 'str'}
+        Specify which arrays to initialize, and desired order.
+        If omitted, returned arrays are ('float', 'int', 'str').
+
+    Returns
+    -------
+    arrays : list of numpy.ndarrays
+        The desired SAC header arrays.
+
+    """
+    if len(args) == 0:
+        args = ('float', 'int', 'str')
+    #http://stackoverflow.com/a/13052254/745557
+    out = []
+    for itype in args:
+        if itype == 'float':
+            # null float header array
+            # XXX: why not init native byte order?
+            hf = np.ndarray(70, dtype=native_str('<f4'))
```

Comment:
Picking up on the todo item: Why not just use the native byte order and convert to it in case it differs from what is stored in the SAC file?

Did anyone actually ever run ObsPy on a big endian architecture?